src/FSharpVSPowerTools.Logic/GoToDefinitionFilter.fs

```diff
@@ -421,10 +432,21 @@ type GoToDefinitionFilter(textDocument: ITextDocument,
                 match symbolResult with
                 | Some (_, _, _, _, FSharpFindDeclResult.DeclFound _) 
                 | None ->
-                    // Run the operation on UI thread since continueCommandChain may access UI components.
-                    do! Async.SwitchToContext uiContext
-                    // Declaration location might exist so let Visual F# Tools handle it. 
-                    return continueCommandChain()
+                    // no FSharpSymbol found, here we look at #load directive
```

Comment:
The logic here is a bit convoluted. If we modify getDocumentState to return:
```fsharp
type Result =
| DeclFound of ...
| DeclNotFound of ...
| NoResult of filePath:string * project:IProjectProvider
```

then we can remove an extra call to `getCurrentFilePathProjectAndDoc`.

Moreover, `Async.RunSynchronously` could easily be removed (We generally avoid it in the code base).